### PR TITLE
feat(FileUploader): Disable uploads on public access

### PR DESCRIPTION
**Pull request by ronelvcabrera**
https://github.com/newmatik/newmatik/issues/1356

newmatik 1356 fix 2

### CODE REVIEW

**Comment by KanchanChauhan** [comment]
I think we can just make this by default checked. No need to write more custom code. And in case of files which needs to be public like blogger images and KB cover image, it will be easy to set them public.
https://github.com/newmatik/frappe/blob/version-12/frappe/public/js/frappe/file_uploader/FileUploader.vue#L50

**Comment by creamdory** [comment]
> @ronelvcabrera @creamdory I think we can just make this by default checked. No need to write more custom code. And in case of files which needs to be public like blogger images and KB cover image, it will be easy to set them public.
> https://github.com/newmatik/frappe/blob/version-12/frappe/public/js/frappe/file_uploader/FileUploader.vue#L50

@KanchanChauhan https://github.com/newmatik/newmatik/issues/1356#issuecomment-520822201

**Comment by creamdory** [comment]
@dottenbr @KanchanChauhan because of issue https://github.com/newmatik/newmatik/issues/2307, It would be best if the design is this way:

1) There is a global `System Setting` to force upload to private as the design above.
2) There is a `Doctype`-specific checkbox to `Allow Upload to Public` for that Doctype.

@KanchanChauhan the purpose of the forced upload and not defaulting it only to checked is because we want to make sure the User cannot edit it. This is what happen in our first implementation resulting it to be custom to hide the checkbox in current version.

**Comment by ronelvcabrera** [comment]
@creamdory Unable to tag Project for this PR. I dont know why.


Screen Shot 2019-11-07 at 6 24 31 PM

**Comment by creamdory** [comment]
@ronelvcabrera it is in recent tab. I already tagged it.

feat(FileUploader): System Setting to enable or disable uploads on public access by ronelvcabrera · Pull Request #27 · newmatik… 2019-11-08 09-56-48

**Comment by KanchanChauhan** [comment]
@creamdory @ronelcabrera As you are aware we are re-doing it. It was done before Stavros but not accepted by Frappe, we updated and lost our code. Now we are doing it again. So this is an endless cycle every time we update. If the team is ok with it, I am ok too.

File: frappe/public/js/frappe/file_uploader/index.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ export default class FileUploader {
 		} else {
 			this.wrapper = wrapper.get ? wrapper.get(0) : wrapper;
 		}
-
+		let private_only = (parseInt(frappe.boot.sysdefaults.is_private_only)) ? true:false;
```
Comment (KanchanChauhan) [inline frappe/public/js/frappe/file_uploader/index.js]: I am just wondering, we have checked the box (Make all attachments private) and disabled it, which makes adding a checkbox in System Settings obsolete. Even if we uncheck it, nothing will change.

File: frappe/public/js/frappe/file_uploader/FileUploader.vue [hunk 2]
```diff
@@ -220,7 +228,10 @@ export default {
 			let files = Array.from(file_array)
 				.filter(this.check_restrictions)
 				.map(file => {
-					let is_image = file.type.startsWith('image');
+					let is_private = !file.type.startsWith('image');
```
Comment (KanchanChauhan) [inline frappe/public/js/frappe/file_uploader/FileUploader.vue]: Why do we need this change?

**Comment by ronelvcabrera** [comment]
> @creamdory @ronelcabrera As you are aware we are re-doing it. It was done before Stavros but not accepted by Frappe, we updated and lost our code. Now we are doing it again. So this is an endless cycle every time we update. If the team is ok with it, I am ok too.

Yes as discussed, will update that this is only applied to our repo only. Will update once done.

**Comment by ronelvcabrera** [comment]
@KanchanChauhan @creamdory Updates made. Tested. All good. Kindly review. Thank you.

**Pull request merged**